daemon/containerd/image_list.go

```diff
@@ -174,26 +186,60 @@ func (i *ImageService) Images(ctx context.Context, opts imagetypes.ListOptions)
 				return nil
 			}
 
-			image, chainIDs, err := i.singlePlatformImage(ctx, contentStore, tagsByDigest[img.RealTarget.Digest], img, opts, allContainers)
+			conf, err := img.Config(ctx)
 			if err != nil {
 				return err
 			}
 
-			summaries = append(summaries, image)
+			var dockerImage dockerspec.DockerOCIImage
+			if err := readConfig(ctx, contentStore, conf, &dockerImage); err != nil {
+				return err
+			}
+
+			presentImages = append(presentImages, tempImage{
+				img:           img,
+				indexPlatform: img.Target().Platform,
+				dockerImage:   &dockerImage,
+			})
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		if len(presentImages) == 0 {
+			// TODO we should probably show *something* for images we've pulled
+			// but are 100% shallow or an empty manifest list/index
+			// ("tianon/scratch:index" is an empty example image index and
+			// "tianon/scratch:list" is an empty example manifest list)
```

Comment:
Ah, yes, was about to ask if we could test this. As these are minimal images, perhaps we could include them in our "frozen images"?

I'm fine with doing that in a follow-up if it's complicated though.

Reply:
Not sure if it makes sense to handle this right now.

1. It's impossible to pull such image, you can only load it via `docker load`.
2. Even if it is loaded, there's nothing that can be done with that image (except removing/pushing it).